libnetwork/datastore/datastore.go

```diff
@@ -109,15 +102,15 @@ func Key(key ...string) string {
 
 // New creates a new Store instance.
 func New(cfg ScopeCfg) (*Store, error) {
-	if cfg.Client.Path == "" {
+	if cfg.Path == "" {
 		cfg = DefaultScope("")
```

Comment:
Was looking at this code, and this is actually a weird construct; if we don't set a Path in the config, we IGNORE everything else, and just overwrite the whole `cfg`.

Wondering if the better signature would be to have `New(path string, cfg ScopeCfg)`

- `cfg` could be a pointer, so that `nil` can be passed to have the constructor use defaults
- export the default path constant

The latter one should be changed; currently we have a `defaultPrefix` const, which is only used in 1 location, and _always_ suffixed with `local-kv.db`; we could make that const `DefaultPath` (e.g.) and make passing a path a _required_ option 🤔 https://github.com/moby/moby/blob/98fecb0d54994b3b0c645b9b721651b03198bf9e/libnetwork/datastore/datastore.go#L77-L83